src/librustc_codegen_ironox/ir/basic_block.rs

```diff
@@ -16,32 +10,30 @@ pub struct BasicBlock(pub usize, pub usize);
 
 /// A basic block.
 #[derive(Debug, PartialEq)]
-pub struct BasicBlockData {
+pub struct OxBasicBlock {
     /// The label of the basic block.
     pub label: String,
+    /// The index of the basic block in the parent function.
+    pub idx: usize,
```

Comment:
Does the LLVM backend store this too? Can it not be computed as needed from the basic block backing array, or is that too expensive?

I guess it's memory vs. speed.

Reply:
Let's roll with what you have for now.